### ISSUE TITLE
Build docker image excluding the artifact jar itself without any warning

### ISSUE THREAD

**Issue by Journerist** [fabric8io/docker-maven-plugin#1121]
### Description

For a very slim cassandra container we don't need to include and build the jar of the maven subproject. Therefore, we skip the usage of the maven-jar-plugin and no artifact jar is built. Unfortunately, we receive a warning and we would like to get rid of it.

```
[WARNING] Cannot include project artifact: com.mypackage:jar:TRUNK-SNAPSHOT; it doesn't have an associated file or directory.
[WARNING] The following patterns were never triggered in this artifact inclusion filter:
```

### Info

* d-m-p version : 0.26.0
* Maven version (`mvn -v`) :
```
Maven home: /usr/local/Cellar/maven/3.5.4/libexec
Java version: 1.8.0_181, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/jdk1.8.0_181.jdk/Contents/Home/jre
Default locale: en_DE, platform encoding: UTF-8
OS name: "mac os x", version: "10.13.6", arch: "x86_64", family: "mac"
```

I know it's a bit dirty to have a maven submodule that only includes dependencies to its docker container without having own java files. Anyhow, I would like to get rid of the warning to have a clean build.

Thanks in advance

**Comment by rhuss** [comment]
I know this warning is super tedious and there is an issue that you need to have a `package` before a `docker:build` so that a jar would be picked. However, this warning comes directly from the assembly builder (not our code), so I wonder how your build looks like. Are you able to post a sample project? Because if you don't need an assembly for specifying the content of your Docker image, then it should be possible to fix this. However, I would need to be able to reproduce it.

**Comment by fxnn** [comment]
For me, this occurs with `pom` packaging. 

Reasoning: In our multimodule project, we use to have different Maven modules for different outcomes (`war`, `it`, `docker` etc.). Therefore, there simply _is_ no project artifact to be attached. Using the assembly, we simply source the required WAR and other files using `<dependencySets>`.

Besides, thanks for the greatly designed plugin :)

**Comment by rhuss** [comment]
Yeah, please open a new issue for issues with pom type packaging and bogus builds. I think we added something that the plugin should skip build on pom packaging projects, but not sure.
